site_scons/site_tools/go_builder.py

```diff
@@ -49,6 +49,20 @@ def _scan_go_file(node, env, _path):
     return includes
 
 
+def get_go_version(output):
+    """Capture only the number, ignoring optional rc"""
+    one_part = '([0-9]+)(?:rc[0-9]+)?'
+    return '.'.join(re.findall(fr'(?:go)?{one_part}\.{one_part}\.{one_part}', output)[0])
+
+
+def test_go():
+    """Quick unit test"""
+    assert "1.20.10" == get_go_version("go version go1.20.10 linux/amd64")
+    assert "1.2.3" == get_go_version("go version go1.2.3 Linux/amd64")
+    assert "1.20.10" == get_go_version("go version go1.20.10-daos linux/amd64")
+    assert "1.20.10" == get_go_version("go version go1.20rc2.10 linux/amd64")
```

Comment:
Are `go1.20rc2.10` and `go1.20.10` actually the same thing given that this function is returning them as equivalent?  If not will this ever be a problem with regard to minimum version matching?

Reply:
This check appears to be there to enforce a minimum go compiler version so that we can use language features without further checking for them in the code, therefore I would expect those two to be the same.

Reply:
I would also like to know that, in fact if you read the ticket you can see I've been asking exactly the same question.

The ticket relates to the daos_build test on Leap, it appears that's whats happening is that daos is built into an RPM and is happy with the scons version at that point, then the test tries to build daos over dfuse for testing and at that point the build of daos is failing with a go version that it can not parse.  I do not know the precise version of go that is being used for the test as the build did not log it (that's one change in this PR) and also the test is no longer failing, presumably as there's been another update to a non-rc version.

I neither know why Leap15 build and test were using different go versions, which the exact version was that was failing or what changed to enable the test to work.

I assume that it was using a rc build of 1.20 but that's purely an assumption, I just don't have more to go on despite looking for the information.

The test is passing again now, this PR should allow it to continue to work if a rc version is encountered again, plus as I say it'll log the actual version string this time so if this ever becomes an issue again we have more to work with.

Reply:
Should we hold off on landing this then until we have a test case in hand that we can further understand the situation and answer some of the open questions?

My concern is just landing something that provides a false report on the **go** version being used in cases where one of these mysterious _rc_ releases shows up.